pkg/api/universal/conversation.go

```diff
@@ -442,26 +449,54 @@ func (a *Universal) ConverseAlpha2(ctx context.Context, req *runtimev1pb.Convers
 				}
 			}
 			if !toolNameFound {
-				err = messages.ErrConversationInvalidParams.WithFormat(req.GetName(), "tool choice not found. Must be 'auto', 'none', 'required', or a specific tool name matching the tools available to be used")
+				err = messages.ErrConversationInvalidParams.WithFormat(req.GetName(), "tool choice selected was not found. Must be 'auto', 'none', 'required', or a specific tool name matching the tools available to be used")
 				a.logger.Debug(err)
 				return nil, err
 			}
 		}
 	}
 
-	request.ToolChoice = &toolChoice
+	if toolChoice != "" {
+		request.ToolChoice = &toolChoice
+	}
 
 	if tools := req.GetTools(); tools != nil {
 		availableTools := []llms.Tool{}
 		for _, tool := range tools {
 			switch t := tool.GetToolTypes().(type) {
 			case *runtimev1pb.ConversationTools_Function:
+				// only type, required, and properties are supported in most LLMs for jsonschema attributes
+				params := map[string]any{
+					"type": "object",
+				}
+				for key, param := range t.Function.GetParameters() {
+					fmt.Printf("key: %s, value: %s\n", key, param.String())
+					if key == "required" {
```

Comment:
this statements would probably not be needed after https://github.com/dapr/dapr/pull/8938 and https://github.com/dapr/dapr/pull/8939